src/CMakeLists.txt

```diff
@@ -259,30 +264,26 @@ elseif (MULTI_THREAD)
     set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "/MT ${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
 endif ()
 
-if (EMSCRIPTEN)
-    set(gmp_install_prefix ${CMAKE_CURRENT_BINARY_DIR}/gmp-root)
-    # The plethory of configure arguments only makes sure that gmp thinks sizeof(mp_limb_t) == sizeof(long) == 4
-    ExternalProject_Add(
-            gmp
-            URL https://gmplib.org/download/gmp/gmp-6.1.1.tar.bz2
-            URL_HASH SHA256=a8109865f2893f1373b0a8ed5ff7429de8db696fc451b1036bd7bdf95bbeffd6
-            BUILD_IN_SOURCE 1
-            CONFIGURE_COMMAND emconfigure ./configure "CC_FOR_BUILD=gcc" "CCAS=gcc -c" "CFLAGS=-m32 -DPIC ${CFLAGS_EMSCRIPTEN}" --host=x86_64-pc-linux-gnu --build=i686-pc-linux-gnu --disable-assembly --prefix=${gmp_install_prefix}
-            BUILD_COMMAND emmake make -j4
-            INSTALL_COMMAND make install
-    )
-    add_library(lean_external_deps INTERFACE)
-    target_link_libraries(lean_external_deps INTERFACE ${gmp_install_prefix}/lib/libgmp.so)
-    include_directories(lean_external_deps INTERFACE ${gmp_install_prefix}/include)
-    add_dependencies(lean_external_deps gmp)
-    set(EXTRA_LIBS lean_external_deps)
+if ("${CMAKE_SYSTEM_NAME}" MATCHES "Emscripten")
+  set(gmp_install_prefix ${CMAKE_CURRENT_BINARY_DIR}/gmp-root)
+  ExternalProject_Add(
+    gmp
+    URL https://gmplib.org/download/gmp/gmp-6.2.1.tar.bz2
+    URL_HASH SHA256=eae9326beb4158c386e39a356818031bd28f3124cf915f8c5b1dc4c7a36b4d7c
+    BUILD_IN_SOURCE 1
+    CONFIGURE_COMMAND emconfigure ./configure "CFLAGS=${EMSCRIPTEN_SETTINGS}" --host=wasm32-unknown-emscripten --disable-assembly --prefix=${gmp_install_prefix}
+    BUILD_COMMAND emmake make -j4
+    INSTALL_COMMAND emmake make install
+  )
+  include_directories(${gmp_install_prefix}/include)
+  set(EXTRA_LIBS ${EXTRA_LIBS} ${gmp_install_prefix}/lib/libgmp.a)
 else()
-    # GMP
-    find_package(GMP 5.0.5 REQUIRED)
-    include_directories(${GMP_INCLUDE_DIR})
-    set(COPY_LIBS ${COPY_LIBS} ${GMP_LIBRARIES})
```

Comment:
Removed `COPY_LIBS` as it doesn't seem to have any role other than adding to `EXTRA_LIBS`